### PR TITLE
building: trusted applications: get client identity

### PR DESCRIPTION
**Pull request by etienne-lms**
Describe a bit TEE login methods OP-TEE implements and how a TA can sue it.
I've created this change based on https://github.com/OP-TEE/optee_os/issues/6436.

### CODE REVIEW

**Comment by jbech-linaro** [comment]
Commit message s/sue/use/.

File: building/trusted_applications.rst [hunk 1]
```diff
@@ -338,6 +338,68 @@ sequence:
         }
     }
 
+
+Identifying TA's client
+***********************
+
+GP TEE specification is mades so that TEE sessions, once established, are
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:345]: s/mades/made/
But, an alternative could be:
"The GP TEE specification is designed to ensure that TEE sessions are reliable once created."

File: building/trusted_applications.rst [hunk 2]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+reliable. A TA instance can identify its client login method when session is
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:346]: s/when session/when a session/

File: building/trusted_applications.rst [hunk 3]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+opened. Based on the login, the TA can accpet of not the session. A TA can
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:347]: s/accpet/accept/

But the sentence itself is probably wrong, I don't get what you intend to say, can you perhaps rephrase it?

File: building/trusted_applications.rst [hunk 4]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+get its client identity with TEE Internal Core API function
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:348]: s/with TEE/with the TEE/

File: building/trusted_applications.rst [hunk 5]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+        case TEE_LOGIN_TRUSTED_APP:
+            return login_ta(&identity.uuid, tee_session);
+        default:
+            return TEE_ERROR_ACCESS_DENIED;
+        }
+    }
+
+The value of the UUID found in ``identity.uuid`` depends on the login method:
+
+    - When client is a TA, ``identity.login`` is ``TEE_LOGIN_TRUSTED_APP`` and
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:383]: s/When client/When the client/

File: building/trusted_applications.rst [hunk 6]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+      ``identity.uuid`` is the client TA UUID;
+
+    - When non-secure client uses ``TEE_LOGIN_PUBLIC`` or
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:386]: s/When non-secure/When the non-secure/

File: building/trusted_applications.rst [hunk 7]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+      ``TEE_LOGIN_REE_KERNEL``, the UUID is not used. By convention, Linux
+      kernel and U-Boot both set nil UUID (all zeroes).
+
+    - when non-secure client uses ``TEE_LOGIN_USER`` or ``TEE_LOGIN_GROUP``,
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:390]: s/When non-secure/When the non-secure/

File: building/trusted_applications.rst [hunk 8]
```diff
@@ -338,6 +338,68 @@ sequence:
...
+      the UUID is generated from the UUIDv5 namespace derivation of a user ID
+      tag (``uid=%x\0``) or a group ID tag (``gid=%x\0``) in
+      ``tee_client_uuid_ns`` namespace (below). The derivation is performed
+      by the Linux kernel that verifies that client's UID/GID is genuine,
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:394]: s/that client's/that the client's/

**Review by etienne-lms — COMMENTED** [review]
comments addressed (but the one related to the commit message).

**Review by jbech-linaro — APPROVED** [review]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: building/trusted_applications.rst [hunk 9]
```diff
@@ -338,6 +338,69 @@ sequence:
...
+    - When the client is a TA, ``identity.login`` is ``TEE_LOGIN_TRUSTED_APP``
+      and ``identity.uuid`` is the client TA UUID;
+
+    - When the non-secure client uses ``TEE_LOGIN_PUBLIC`` or
+      ``TEE_LOGIN_REE_KERNEL`` method, the UUID is not used. By convention,
+      Linux kernel and U-Boot both set nil UUID (all zeroes).
+
+    - When the non-secure client uses ``TEE_LOGIN_USER`` or ``TEE_LOGIN_GROUP``
+      method, the UUID is generated from the UUIDv5 namespace derivation of a
+      user ID tag (``uid=%x\0``) or a group ID tag (``gid=%x\0``) in
```
Comment (etienne-lms) [inline building/trusted_applications.rst:393]: error: the terminal `\0` is excluded from the tag. I'll also add double-quote `"` to highlight it's a string based ID:
```patch
-      user ID tag (``uid=%x\0``) or a group ID tag (``gid=%x\0``) in
+      user ID tag (``"uid=%x"``) or a group ID tag (``"gid=%x"``) in

**Comment by etienne-lms** [comment]
fixup commit squashed.

**Pull request merged**